core/src/main/java/org/elasticsearch/plugins/PluginManager.java

```diff
@@ -469,6 +490,11 @@ Path newDistroFile(Environment env) throws IOException {
             return Files.createTempFile(env.tmpFile(), name, ".zip");
         }
 
+        Tuple<URL, Path> newChecksumUrlAndFile(Environment env, URL originalUrl, String suffix) throws IOException {
+            URL newUrl = new URL(originalUrl.toString() + "." + suffix);
```

Comment:
any chance we can get rid of URL instead use URI? it might be baked in then I think we should open a followup to clean this up